flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1101,11 +1107,70 @@ class ExprLowering {
         si.box);
   }
 
+  fir::ExtendedValue genArrayCoorOp(const Fortran::lower::SymbolBox &si,
+                                    const Fortran::evaluate::ArrayRef &aref) {
+    auto loc = getLoc();
+    auto addr = si.getAddr();
+    auto arrTy = fir::dyn_cast_ptrEleTy(addr.getType());
+    auto eleTy = arrTy.cast<fir::SequenceType>().getEleTy();
+    auto refTy = builder.getRefType(eleTy);
+    auto idxTy = builder.getIndexType();
+    auto genWithShape = [&](const auto &arr) -> mlir::Value {
+      mlir::Type shapeType =
+          fir::ShapeType::get(builder.getContext(), arr.getExtents().size());
+      auto shapeOp =
+          builder.create<fir::ShapeOp>(loc, shapeType, arr.getExtents());
+
+      llvm::SmallVector<mlir::Value, 8> arrayCoorArgs;
+      // arrayCoorArgs.push_back(shapeOp);
+      for (const auto &sub : aref.subscript()) {
+        auto subVal = genComponent(sub);
+        if (auto *ev = std::get_if<fir::ExtendedValue>(&subVal)) {
+          if (auto *sval = ev->getUnboxed()) {
+            auto sub = builder.createConvert(loc, idxTy, *sval);
+            arrayCoorArgs.push_back(sub);
+          } else {
+            TODO();
+          }
+        }
```

Comment:
Should there also be an `else {TODO()}` here for RangeBox or is it unreachable ?